src/commands/createNewProject.ts

```diff
@@ -8,25 +8,89 @@ import * as path from 'path';
+    } else {
+        functionAppPath = workingDirectory;
+    }
+
+    const languages: Pick[] = [
+        new Pick(TemplateLanguage.JavaScript),
+        new Pick(TemplateLanguage.Java)
+    ];
+    const language: Pick = await ui.showQuickPick(languages, localize('azFunc.selectFuncTemplate', 'Select the language of your function project'));
+
+    let javaTargetPath: string = '';
+    switch (language.label) {
+        case TemplateLanguage.Java:
+            const groupIdPlaceHolder: string = localize('azFunc.java.groupIdPlaceholder', 'groupId');
+            const groupIdPrompt: string = localize('azFunc.java.groupIdPrompt', 'Provide value of "groupId"');
+            const groupId: string = await ui.showInputBox(groupIdPlaceHolder, groupIdPrompt, false);
+
+            const artifactIdPlaceHolder: string = localize('azFunc.java.artifactIdPlaceholder', 'artifactId');
+            const artifactIdprompt: string = localize('azFunc.java.artifactIdPrompt', 'Provide value of "artifactId"');
+            const artifactId: string = await ui.showInputBox(artifactIdPlaceHolder, artifactIdprompt, false);
+
+            const versionPlaceHolder: string = localize('azFunc.java.versionPlaceHolder', '1.0-SNAPSHOT');
+            const versionPrompt: string = localize('azFunc.java.versionPrompt', 'Provide value of "version"');
+            const version: string = await ui.showInputBox(versionPlaceHolder, versionPrompt, false, undefined, '1.0-SNAPSHOT');
+
+            const packagePlaceHolder: string = localize('azFunc.java.packagePlaceHolder', groupId);
+            const packagePrompt: string = localize('azFunc.java.packagePrompt', 'Provide value of "package"');
+            const packageName: string = await ui.showInputBox(packagePlaceHolder, packagePrompt, false, undefined, groupId);
+
+            const appNamePlaceHolder: string = localize('azFunc.java.appNamePlaceHolder', `${artifactId}-${Date.now()}`);
+            const appNamePrompt: string = localize('azFunc.java.appNamePrompt', 'Provide value of "appName"');
+            const appName: string = await ui.showInputBox(appNamePlaceHolder, appNamePrompt, false, undefined, `${artifactId}-${Date.now()}`);
+
+            // Use maven command currently, will change to function CLI when the function CLI init project command is ready.
+            await FunctionsCli.createNewProject(
+                outputChannel,
+                functionAppPath,
+                language.label,
+                'archetype:generate',
+                '-DarchetypeGroupId="com.microsoft.azure"',
+                '-DarchetypeArtifactId="azure-functions-archetype"',
+                `-DgroupId="${groupId}"`,
+                `-DartifactId="${artifactId}"`,
+                `-Dversion="${version}"`,
+                `-Dpackage="${packageName}"`,
+                `-DappName="${appName}"`,
+                '-B' // in Batch Mode
+            );
+
+            functionAppPath = path.join(functionAppPath, artifactId);
```

Comment:
This is because java function project that generated by the [Maven plugin](https://github.com/Microsoft/azure-maven-plugins/tree/master/azure-functions-maven-plugin) is different from JS function in terms of the folder structure.

For example, this is what it looks like when we use ```func init``` to create a function project
![image](https://user-images.githubusercontent.com/6193897/32471733-c2a91d94-c399-11e7-9d77-351fe7ee6c78.png)

And this is what it looks like to use maven function archetype:
![image](https://user-images.githubusercontent.com/6193897/32471864-72258226-c39a-11e7-912e-121c03b85dbd.png)

So the reason why join the path when it's a java function project is that we want to open the folder that contains the function related files (```host.json```, ```local.settings.json```, etc.) also the ```pom.xml``` which is java specific.

Reply:
The above examples look like the same folder structure to me. It's just that in the top one the user selected `D:\` as the functionAppPath and in the bottom one the user selected `D:\myfunction` as the functionAppPath. (I understand that `func init` and `mvn` deal with this slightly differently, but the user doesn't care about that internal detail. They just want the same behavior in both scenarios)

With your current code, I end up with an extra folder every time:
<img width="609" alt="screen shot 2017-11-06 at 6 13 59 pm" src="https://user-images.githubusercontent.com/11282622/32473862-8385ec76-c31e-11e7-94d7-37e59f24c436.png">

I'm never going to put anything else in 'myFunctionProject'. It seems like rather than asking for the artifactId, you should just infer it from the parent directory name

Reply:
Yes, what I mean is that the ```myfunction``` is the extra folder generated by the maven archetype. (In your case it's the folder: ```artifactId```). The functionAppPath user selected in both senario is the same. Unfortunately, the maven archetype will always create the extra folder for the user. How about treat it as you said, infer artifactId from the parent directory name, without asking user for it. And for Java function project, we still keep the path join logic here to ensure that the extension will open the right base folder for function project?

Reply:
What if you run the maven command from the parent folder - with the folder name as the artifactId? Wouldn't that solve the issue?

Reply:
No, the maven CLI will fail when we do that. 😢

Reply:
Fail with what error?

Reply:
Here is the screenshot:
![image](https://user-images.githubusercontent.com/6193897/32476181-a4690ff8-c3b0-11e7-89b8-0ab6827318ee.png)

Reply:
BTW, the error msg ouput by Maven is not that meaningful.

Reply:
Alright - please file an issue for this. I don't see an obvious solution, but I still think it's worth further investigation later

Reply:
Yes. I agree with you. I'll file a issue later.